Add extend-access-ws web service sample for OIG/WebLogic

mgere-oracle opened the pull request:
WebService which is complementary to #361.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+
+		String wlDomainDirectory = System.getProperty("domain.home");
+		configurationsFileInput = new FileInputStream(
+				java.nio.file.Paths.get(wlDomainDirectory, "config", "extendaccessws.properties").toString());
```
Hidden imports in the code are not a good coding practice. Add this class to the import section.

mgere-oracle replied:
Ouch indeed. Not sure how my linter let this through. Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+		Properties prop = new Properties();
+		InputStream configurationsFileInput = null;
+
+		String wlDomainDirectory = System.getProperty("domain.home");
```
What if "domain.home" is empty?

mgere-oracle replied:
Shouldn't happen if one deploys on a WebLogic node, but fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+			logger.log(Level.WARNING, "Unable to close properties file.");
+		}
+
+		String wsOIGNodeURL = prop.getProperty("wsOIGNodeURL");
```
What if this or one of the following properties is not present in the properties files? Any reasonable defaults or error messages?

mgere-oracle replied:
No defaults, since these will vary depending on the deployment environment, but I'm validating for the presence of the fields now. Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+
+		try {
+			// connect to OIM using the oimClient service
+			if (oimClient == null)
```
Surround every block even single line blocks with curly braces. One famous Apple TLS bug was because of missing curly braces.

mgere-oracle replied:
Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+		env.put(OIMClient.JAVA_NAMING_FACTORY_INITIAL, "weblogic.jndi.WLInitialContextFactory");
+		env.put(OIMClient.JAVA_NAMING_PROVIDER_URL, wsExtendAccessNodeURL);
+		Context ctx = new InitialContext(env);
+		DataSource ds = (javax.sql.DataSource) ctx.lookup("jdbc/operationsDB");
```
Hidden imports in the code are not a good coding practice. Add this class to the import section.

mgere-oracle replied:
Fixed, as above.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+		Properties prop = new Properties();
+		InputStream configurationsFileInput = null;
+
+		String wlDomainDirectory = System.getProperty("domain.home");
```
See comment for line 110.

mgere-oracle replied:
Fixed, as above.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,348 @@
+			logger.log(Level.WARNING, "Unable to close properties file.");
+		}
+
+		String wsExtendAccessNodeURL = prop.getProperty("wsExtendAccessNodeURL");
```
See comment for line 121.

mgere-oracle replied:
Fixed, as above.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/README.md:
```diff
@@ -0,0 +1,70 @@
+
+The WebService is build using the JAX-RS specification and API.
+
+As such, the following jar files are required as dependencies and need to be used during the build process:
```
Please add a sample command line for compiling the code.

mgere-oracle replied:
Typically one would use Ant/Maven here. I've added a line to that extent with a manual war file build example as well.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/WebContent/wsdl/ExtendAccessWSv1.wsdl:
```diff
@@ -0,0 +1,91 @@
+   </element>
+  </schema>
+ </wsdl:types>
+
```
Remove empty lines.

mgere-oracle replied:
Done. Note that this is an auto-generated file, only provided as an example.

AlexanderHodicke commented:
@mgere-oracle Please see olaf Feedback.

mgere-oracle commented:
> @mgere-oracle Please see olaf Feedback.

I have addressed all the points raised by @oheimburger. Please let me know if there's anything left pending from my side.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/WebContent/WEB-INF/ExtendAccessWSv1Service/com/oracle/samplews/deploy.wsdd.bak:
Backup files should not be pushed to github.

mgere-oracle replied:
Sorry about that. The entire WebContent directory was auto-generated so I did not catch it. Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,368 @@
+			logger.log(Level.SEVERE, "Severe exception encountered: ", e);
+		} finally {
+			try {
+				if (resultSet != null && !resultSet.isClosed())
```
Use curly braces for single line blocks.

mgere-oracle replied:
Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,368 @@
+				logger.log(Level.WARNING, "Error on RS termination: ", e);
+			}
+			try {
+				if (preparedStatement != null && !preparedStatement.isClosed())
```
Use curly braces for single line blocks.

mgere-oracle replied:
Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java:
```diff
@@ -0,0 +1,368 @@
+				logger.log(Level.WARNING, "Error on PS termination: ", e);
+			}
+			try {
+				if (dbConnection != null && !dbConnection.isClosed())
```
Use curly braces for single line blocks.

mgere-oracle replied:
Fixed.
